app/models/concerns/patient_reportable.rb

```diff
@@ -35,8 +43,9 @@ module PatientReportable
       end
     end
 
-    def self.ltfu_time_ago(date)
-      (date.to_date - LTFU_TIME).end_of_month
+    # Returns a plain timestamp to be interpreted in reporting time, for use with materialized views
+    def self.reporting_timestamp(time)
+      time + time.utc_offset
```

Comment:
I switched my console to IST to try and play out what will happen here, assuming the reporting date is today.

```
31] pry(main)> Time.zone = "Asia/Kolkata"
=> "Asia/Kolkata"
[32] pry(main)> time = Time.current
=> Thu, 03 Jun 2021 21:43:12 IST +05:30
[33] pry(main)> offset = time + time.utc_offset
=> Fri, 04 Jun 2021 03:13:12 IST +05:30
```

I _think_ that is what we want, but it still reads really weird to _add_ the utc_offset to our application time, which is already in the offset time.  Is there a better way we could express this?

Reply:
Yeah, it's weird because we add an _extra_ 5.5 hours, because we know Rails is going to accidentally _subtract_ 5.5 hours when it runs the query.

In any case, #2567 is the simpler approach.